layers/meta-balena-genericx86/recipes-bsp/grub/grub-conf/grub.cfg_internal_template

```diff
@@ -22,52 +23,44 @@ if [ ${upgrade_available} = 1 ] ; then
  fi
 fi
 
-# Append to grub_cmdline the UUID value for all the OS partitions. These UUIDs
-# are stored on the boot partition (/os_fs_uuids.txt). This file contains a
-# line for each partition UUID. For example: "uuid_boot=FFFF-FFFF".
-# Also, the assumed label to kernel parameter mapping is:
-#   resin-boot -> uuid_boot
-#   resin_rootA -> uuid_roota
-#   resin_rootB -> uuid_rootb
-#   resin_state -> uuid_state
-#   resin_data -> uuid_data
-search --set=boot --label resin-boot
-source (${boot})/os_fs_uuids.txt
-if [ x${uuid_boot} != x ]; then
- # If one is available we assume all the rest are as well.
- grub_cmdline="${grub_cmdline} uuid_boot=${uuid_boot}"
- grub_cmdline="${grub_cmdline} uuid_roota=${uuid_roota}"
- grub_cmdline="${grub_cmdline} uuid_rootb=${uuid_rootb}"
- grub_cmdline="${grub_cmdline} uuid_state=${uuid_state}"
- grub_cmdline="${grub_cmdline} uuid_data=${uuid_data}"
-fi
+function get_root_uuid {
+ # Extract drive, partition type and partition number from boot drive in grub format
+ regexp -s 1:bdrive -s 2:pscheme -s 3:bpart '^(hd.*),(msdos|gpt)(.*)' $root
```

Comment:
Looks like `bpart` is not used?

Reply:
You're right, but given that it's parsing the boot drive I just thought I'd extract all the info. Should I remove it?

Reply:
Well, since we do not make use of it, let's remove it. This would mean changing the regex

Reply:
hi @floion it's done now